File: C_Range_Operation.py

# import sys


# input = sys.stdin.read
# data = input().split()
# t = int(data[0])
# idx = 1

# res = []

# for _ in range(t):
#     n = int(data[idx]); idx += 1
#     a = list(map(int, data[idx:idx + n]))
#     idx += n
    
    

#     prefijo = [0] * (n + 1)
#     for i in range(n):
#         prefijo[i + 1] = prefijo[i] + a[i]
    
#     original = prefijo[n]
#     maximos = 0
    
#     for l in range(n):
#         for r in range(l, n):
#             tam = r - l + 1
#             nuevo = (l + 1) + (r + 1)  
#             improvement = tam * nuevo - (prefijo[r + 1] - prefijo[l])
#             maximos = max(maximos, improvement)
    
#     res.append(str(original + maximos))

# print("\n".join(res))



import sys

input = sys.stdin.read
data = input().split()
t = int(data[0])
idx = 1
results = []

for _ in range(t):
    n = int(data[idx]); idx += 1
    a = [0] + list(map(int, data[idx:idx+n])); idx += n
    

    prefix = [0] * (n + 2)
    for i in range(1, n + 1):
        prefix[i] = prefix[i - 1] + a[i]
    
    original_sum = prefix[n]
    max_gain = 0
    

    B = [0] * (n + 2)
    for l in range(1, n + 1):
        B[l] = l * l - l - prefix[l - 1]
    

    min_B = float('inf')
    for r in range(1, n + 1):
        min_B = min(min_B, B[r])
        current_gain = (r * r + r - prefix[r]) - min_B
        max_gain = max(max_gain, current_gain)
    
    results.append(str(original_sum + max_gain))

print("\n".join(results))

File: C_Range_Operation.cpp
#include <iostream>
#include <vector>
#include <numeric>
#include <algorithm>

void solve() {
    int n;
    std::cin >> n;
    std::vector<long long> a(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> a[i];
    }

    std::vector<long long> prefijo(n + 1, 0);
    for (int i = 0; i < n; ++i) {
        prefijo[i + 1] = prefijo[i] + a[i];
    }

    long long original = prefijo[n];
    long long maximos = 0;

    for (int l = 0; l < n; ++l) {
        for (int r = l; r < n; ++r) {
            long long tam = r - l + 1;
            long long nuevo = (long long)(l + 1) + (r + 1);
            long long subarray_sum = prefijo[r + 1] - prefijo[l];
            long long improvement = tam * nuevo - subarray_sum;
            maximos = std::max(maximos, improvement);
        }
    }

    std::cout << original + maximos << "\n";
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    int t;
    std::cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}
